roles/openshift_health_checker/action_plugins/openshift_health_check.py

```diff
@@ -187,12 +188,29 @@ def run_check(name, check, user_disabled_checks):
     if not is_active:
         return dict(skipped=True, skipped_reason="Not active for this host")
 
+    output_dir = check.get_var("openshift_checks_output_dir", default=None)
+    if output_dir and check.can_write_files:
+        output_dir = "/".join([output_dir, check.get_var("ansible_host"), name])
+        if prepare_output_dir(output_dir):
+            # only give the check an output dir if we could create it
+            check.output_dir = output_dir
```

Comment:
Given the per-host nature of running checks, it made sense to me to have a directory for each host and check that runs. However I don't want a directory for checks that are inactive/skipped/unlisted. So it made sense to me to have the action plugin conditionally create the directories and pass them off to the checks (once determined they're needed). 

The action plugin could also then write a summary of results which would include indications of checks that were inactive or skipped. Basically the action plugin is in charge, the checks just know that they either have a place to write or not.